Analysis/ComputeBtagShapeWeight.py

```diff
@@ -0,0 +1,111 @@
+import ROOT
+import sys
+import numpy as np
+import os
+import math
+import shutil
+import json
+from FLAF.RunKit.run_tools import ps_call
+import re
+
+if __name__ == "__main__":
+    sys.path.append(os.environ["ANALYSIS_PATH"])
+
+import FLAF.Common.Utilities as Utilities
+
+if __name__ == "__main__":
+    import argparse
+    import yaml
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("inputFile", nargs="+", type=str)
+    parser.add_argument("--outFile", required=True, type=str)
+
+    args = parser.parse_args()
+
+    # this code will work on data: in that case all per-file jsons will contain ones
+    # so it'll run the calculation that results in all ratios equals one
+
+    # 1 list files :
+    all_files = [fileName for fileName in args.inputFile]
+
+    dataset_btag_weight_dict = {}
+    for this_json in all_files:
+        with open(this_json, "r") as file:
+            this_json_dict = json.load(file)
+
+            for unc_src, unc_dict in this_json_dict.items():
+                if unc_src not in dataset_btag_weight_dict.keys():
+                    dataset_btag_weight_dict[unc_src] = {}
+
+                this_unc_dict = dataset_btag_weight_dict[unc_src]
+                for lep_cat, lep_cat_dict in unc_dict.items():
+                    if lep_cat not in this_unc_dict.keys():
+                        this_unc_dict[lep_cat] = {}
+
+                    for key, val in lep_cat_dict.items():
+                        if key not in this_unc_dict[lep_cat]:
+                            this_unc_dict[lep_cat][key] = val
+                        else:
+                            this_unc_dict[lep_cat][key] += val
+
+    # merge all JER(Up|Down)_* -> JER(Up|Down), etc
+    # Central does not have up/down, so it's an edge case
+    base_sources = [
+        "_".join(s.split("_")[:-1]) if s != "Central" else s
+        for s in dataset_btag_weight_dict.keys()
+    ]
+    lepton_categories = ["e", "mu", "eE", "eMu", "muMu"]
```

Comment:
At this stage, the code shouldn't depend on channels or how the uncertainty sources are named. Also, why do you need a "base source"? 
You should have a config in HH_bbWW repository where you define all bins. E.g.:
```yaml
- name: eE_2jet
  selection: channelId == Channel::EE && nJets == 2
... 
- name: eE_8jet
  selection: channelId == Channel::EE && nJets >= 8
...
```
Then your json at this point should be something like:
```json
[
 { "bin_name": "eE_2jet",
   "unc_source": "JER"
   "unc_scale": "Up"
   "sum_w_before_btag": 1234.0
   "sum_w_after_btag": 1345.0
 },
...
]
```
So this code simply sums the entries from the input files and returns another JSON with a "ratio" entry added to each bin.